### PR TITLE
Add KAIROSDB format to write_http plugin

### PR DESCRIPTION
**Pull request by beorn-**
Adds KairosDB format for write_http collectd plugin.

see  collectd/collectd#1537

### CODE REVIEW

File: src/utils_format_kairosdb.c [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+    sfree(rates); \
+    return (-ENOMEM); \
+  } \
+  else \
+    offset += ((size_t) status); \
+} while (0)
+
+  if (ds->ds[ds_idx].type == DS_TYPE_GAUGE)
+  {
+    if(isfinite (vl->values[ds_idx].gauge))
```
Comment (rubenk) [inline src/utils_format_kairosdb.c]: space after the if( please.

File: src/utils_format_kairosdb.c [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+                vl->plugin,
+                vl->plugin_instance,
+                vl->type,
+                vl->type_instance,
+                ds->ds[ds_idx].name);
+
+      return (-1);
+    }
+
+    if(isfinite (rates[ds_idx]))
```
Comment (rubenk) [inline src/utils_format_kairosdb.c]: same here.

File: src/write_http.c [hunk 3]
```diff
@@ -487,6 +489,60 @@ static int wh_write_json (const data_set_t *ds, const value_list_t *vl, /* {{{ *
...
+                        pthread_mutex_unlock (&cb->send_lock);
+                        return (-1);
+                }
+        }
+
+        status = format_kairosdb_value_list (cb->send_buffer,
+                        &cb->send_buffer_fill,
+                        &cb->send_buffer_free,
+                        ds, vl, cb->store_rates);
+        if (status == (-ENOMEM))
```
Comment (rubenk) [inline src/write_http.c]: No need for the braces around -ENOMEM I think. Did you copy this code from wh_write_json by any chance? ;)

**Comment by rubenk** [comment]
Looking good, I left some minor nits inline.

File: src/write_http.c (hunk 3 above)
Comment (beorn-) [inline src/write_http.c]: yes, the idea was to stick to what's been done earlier

**Pull request merged**